pkgs/development/python-modules/tree-sitter0_21/default.nix

```diff
@@ -30,6 +31,10 @@ buildPythonPackage rec {
 
   pythonImportsCheck = [ "tree_sitter" ];
 
+  # Needed explicitly for Python >= 3.12 as tree-sitter provides
+  # calls to distutils functions to compile language files
+  propagatedBuildInputs = [ distutils ];
+
```

Comment:
How about making it optional for python >= 3.12?

Reply:
distutils is required for tree-sitter 0.21 to work. It was removed from the core library in Python 3.12, so it's not really an optional dependency here.

Reply:
I suppose this could be something like 

```
if pythonAtLeast "3.12" then [ distutils ] else []
```

if we needed to. But including the distutils works just fine in 3.11. It results in more or less a noop, since distutils is part of the core Python packaging up until 3.12 when it was removed.

Reply:
```nix
propagatedBuildInputs = lib.optionals (pythonAtLeast "3.12") [ distutils ];
```